MetaMask/metamask-extension
# Support translation in background code

## Explanation

Support use of the existing locale files to translate text from background code, and translate the text used in the translation notifications. 

This is required for use cases such as browser notifications, and upcoming approval controller methods that support frontend text arguments.

Rather than duplicate the existing frontend logic, the helper functions have been moved to a new shared module. All related unit tests have been refactored and test coverage extended to cover all related methods.

## Manual Testing Steps

- Verify existing frontend text is translated after changing the language.
- Create a transaction using the test dApp and verify the browser notification content.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
